crates/sparrow-scheduler/src/schedule_count.rs

```diff
@@ -0,0 +1,81 @@
+#[cfg(loom)]
+pub(crate) use loom::sync::atomic::{AtomicUsize, Ordering};
+
+#[cfg(not(loom))]
+pub(crate) use std::sync::atomic::{AtomicUsize, Ordering};
+
+#[repr(transparent)]
+#[derive(Debug, Default)]
+pub(crate) struct ScheduleCount(AtomicUsize);
```

Comment:
This struct provides a way of ensuring we only schedule a task once even if we've queued multiple inputs for it. Needs some documentation. It's intended for use in `Task`, but splitting it out like this seems to make it easier to test.